hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/BasicOmKeyInfo.java

```diff
@@ -254,7 +255,7 @@ public static BasicOmKeyInfo getFromProtobuf(String volumeName,
             basicKeyInfo.getFactor(),
             basicKeyInfo.getEcReplicationConfig()))
         .setETag(basicKeyInfo.getETag())
-        .setIsFile(!keyName.endsWith("/"))
+        .setIsFile(basicKeyInfo.getIsFile())
```

Comment:
If an old client send a request containing BasicKeyInfo to a new OM, the isFile field will not be set and a file would always be identified as not a file.

Reply:
@ptlrs can you please add test case in `hadoop-ozone/dist/src/main/smoketest/compatibility/read.robot`?